README.md

```diff
@@ -224,7 +224,10 @@ High level performance data like Throughput or Percentile Precision can be gener
 ### Concurrency And Number of Workers
 TorchServe exposes configurations that allow the user to configure the number of worker threads on CPU and GPUs. There is an important config property that can speed up the server depending on the workload.
 *Note: the following property has bigger impact under heavy workloads.*
-If TorchServe is hosted on a machine with GPUs, there is a config property called `number_of_gpu` that tells the server to use a specific number of GPUs per model. In cases where we register multiple models with the server, this will apply to all the models registered. If this is set to a low value (ex: 0 or 1), it will result in under-utilization of GPUs. On the contrary, setting to a high value (>= max GPUs available on the system) results in as many workers getting spawned per model. Clearly, this will result in unnecessary contention for GPUs and can result in sub-optimal scheduling of threads to GPU.
+
+**CPU**: there is a config property called `workers` which sets the number of worker threads for a model. The best value to set `workers` to is 1 for CPU inference, otherwise the worker threads will contend with each other causing unncessary context switching. The best way to achieve paralellism on CPU is to then increase the `batch_size` parameter until diminishing returns are observed.  
+
+**GPU**: there is a config property called `number_of_gpu` that tells the server to use a specific number of GPUs per model. In cases where we register multiple models with the server, this will apply to all the models registered. If this is set to a low value (ex: 0 or 1), it will result in under-utilization of GPUs. On the contrary, setting to a high value (>= max GPUs available on the system) results in as many workers getting spawned per model. Clearly, this will result in unnecessary contention for GPUs and can result in sub-optimal scheduling of threads to GPU.
```

Comment:
@msaroufim I think in multi-gpu setting still we have control over to set the number of workers per model. Do we have any data on optimizing number of workers for GPUs.

Reply:
I only have single GPU experiments where if you set num workers more than 1 it just crashes

Reply:
On the num threads that is correct, it's why I abandoned the previous PR on setting torch num threads #1138

Reply:
> I only have single GPU experiments where if you set num workers more than 1 it just crashes

@msaroufim 
this happened in docker? we can sync off-line on this.